src/roi/Roi.ts

```diff
@@ -562,25 +567,33 @@ function getBorders(roi: Roi): { ids: number[]; lengths: number[] } {
 
             if (data[neighbour] !== roi.id && !visitedData.has(neighbour)) {
               visitedData.add(neighbour);
-              surroudingIDs.add(data[neighbour]);
-              let surroundingBorder = surroundingBorders.get(data[neighbour]);
-              if (!surroundingBorder) {
-                surroundingBorders.set(data[neighbour], 1);
+              surroundingID = data[neighbour];
+
+              if (surroundingBorder === undefined) {
+                surroundingBorder = 0;
               } else {
-                surroundingBorders.set(data[neighbour], ++surroundingBorder);
+                ++surroundingBorder;
+              }
```

Comment:
You significantly changed how this value is computed (it probably breaks `getBorders`)

Could you explain?

Reply:
Before the surroundingBorder and surroundingIDs were easily taken from a map and a set respectively.  Now I can't do that because the set will not work with objects the way I want it to. Or maybe I just don't know how to implement it correctly. So I just calculate the values for the object separately and then create and object instead of picking specific values out of a set or a map.